### PR TITLE
feat(FOROME-916): redesign main page sidebar

### PR DESCRIPTION
**Pull request by QSdmitrioul**
Side bar was redesign according figma design from [task](https://quantori.atlassian.net/browse/FOROME-916)

+ FIXED bug
There was a bug with some buttons that open modals. If we were trying to click on them when modal is open, modal didn't close. In this PR it is fixed.

### CODE REVIEW

File: src/components/input-search/input-search.module.css [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+.input-search-big {
+  border-radius: 50vh;
+  width: 100%;
+  padding: 6px 35px 6px 12px;
+  font-size: 14px;
+  line-height: 20px;
+  border: 1px solid theme("colors.grey.blue");
+}
+
+.input-search-big:focus {
```
Comment (AndreyUstyumenko) [inline src/components/input-search/input-search.module.css]: First we work in BEM methodology, so modifyers should be separated with --
Example: .input-search--big
Second, css modules support nesting, so it should be like this:
```
.input-search {
    ...
    &--big {
    }
    &:focus {
    }
}
```

File: src/pages/main/ui/datasets-list/components/dataset-bage.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { FC } from 'react'
+import cn from 'classnames'
+
+interface IXLBageProp {
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/components/dataset-bage.tsx]: IXLBageProp**s**

File: src/pages/main/ui/datasets-list/components/dataset-name.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  className,
+}) => {
+  const datasetRef = useRef<any>(null)
+
+  const isXL = kind?.toLocaleLowerCase() === 'xl'
+
+  const name =
+    isXL && dsName.substring(0, 3).toLocaleLowerCase() === 'xl_'
+      ? dsName.substring(3)
+      : dsName
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/components/dataset-name.tsx]: I think better to use regexp

File: src/pages/main/ui/datasets-list/components/dataset-name.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  const hasTooltip =
+    datasetRef?.current?.getBoundingClientRect().width +
+      datasetRef?.current?.getBoundingClientRect().x >
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/components/dataset-name.tsx]: create a var for datasetRef?.current?.getBoundingClientRect()

File: src/pages/main/ui/datasets-list/components/dataset-name.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          className={cn(
+            kind === null ? 'text-grey-blue' : 'text-white',
+            'text-sm leading-18px relative ml-2 pr-4',
+            {
+              'font-bold': isXL,
+              'font-medium': (isActive || isActiveParent) && !isXL,
+              truncate: !isChildrenVisible,
+              'py-2': !kind,
+            },
+            className,
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/components/dataset-name.tsx]: move cn calculations to a var

File: src/pages/main/ui/datasets-list/components/dataset-time.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      'text-white': isActive,
+      'text-grey-blue': !isActive,
+    })}
+  >
+    <div>{formatDate(time)}</div>
+
+    <div
+      className={cn('border-l my-1 mx-1', {
+        'border-white': isActive,
+        'border-blue-secondary': !isActive,
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/components/dataset-time.tsx]: ternary expression is better

File: src/pages/main/ui/datasets-list/datasets-list-item.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        dirinfoStore.setDsInfo(item as IDirInfoDatasetDescriptor)
+      }
+
+      dirinfoStore.setInfoFrameLink('')
+      dirinfoStore.setActiveInfoName('')
+
+      history.replace(`${Routes.Root}?ds=${isNullKind ? '' : item.name}`)
+    }
+
+    const padding = 18 + level * 16
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/datasets-list-item.tsx]: create vars for 18 and 16. They're not obvious numbers

File: src/pages/main/ui/datasets-list/datasets-list.utils.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import dirinfoStore from '@store/dirinfo'
+import { DatasetsListItem } from '@pages/main/ui/datasets-list/datasets-list-item'
+import { IDirInfoDatasetDescriptor } from '@service-providers/vault-level/vault-level.interface'
+
+export const datasetNameByKey =
+  (level: number = 0) =>
+  // eslint-disable-next-line react/display-name
+  (key: string) => {
```
Comment (AndreyUstyumenko) [inline src/pages/main/ui/datasets-list/datasets-list.utils.tsx]: what's the problem to just make a function with 2 params? Avoid closures when it's not necessary

File: src/pages/ws/ui/control-panel/zone-modals/zone-item.tsx [hunk 9]
```diff
@@ -26,20 +26,22 @@ export const ZoneItem = observer(
...
         text="+"
         refEl={refEl}
         onClick={onClick}
+        onMouseUp={onMouseUp}
         className="text-blue-bright mt-px ml-1"
       />
     )
 
-    const ButtonElementAdd = ({ refEl, onClick }: any) => (
+    const ButtonElementAdd = ({ refEl, onClick, onMouseUp }: any) => (
```
Comment (AndreyUstyumenko) [inline src/pages/ws/ui/control-panel/zone-modals/zone-item.tsx]: fix any

**Comment by AndreyUstyumenko** [comment]
please reorganize the folder structure:
```
main
 - components
   - datasets
       - ...components that are related to datasets
   - selected-dataset
      - ...components that are related to selected-dataset
      - modals
 - page.tsx
```

ui folder should be gone
components that have stores must be in a separate folder

File: src/pages/main/ui/datasets-list/datasets-list.utils.tsx (hunk 8 above)
Comment (QSdmitrioul) [reply src/pages/main/ui/datasets-list/datasets-list.utils.tsx]: I use carrying here, to use produced function in map method

File: src/components/input-search/input-search.module.css (hunk 1 above)
Comment (QSdmitrioul) [reply src/components/input-search/input-search.module.css]: Working on that!

Comment (AlMaQntr) [reply src/components/input-search/input-search.module.css]: I think in BEM traditionally used this style: `some-block__some-element_some-modifier`. Double underscore for element divider and underscore for modifier. Also our webpack configured for transform kebab-case to camelCase in CSS modules

File: src/components/input-search/input-search.tsx [hunk 10]
```diff
@@ -1,32 +1,38 @@
...
+
 import { ChangeEvent, ReactElement } from 'react'
 import cn, { Argument } from 'classnames'
 
 import { Icon } from '@ui/icon'
-import { Input } from '@ui/input'
-import { DecisionTreesResultsDataCy } from '../../src/components/data-testid/decision-tree-results.cy'
+import { DecisionTreesResultsDataCy } from '../data-testid/decision-tree-results.cy'
 
 interface Props {
```
Comment (Lapk1n) [inline src/components/input-search/input-search.tsx]: IInputSearchProps

**Review by QYaroslavTrefilov — APPROVED** [review]

**Review by Lapk1n — COMMENTED** [review]
Screenshot 2022-05-05 at 11 34 28
Screenshot 2022-05-05 at 11 34 42
Maybe we should get back an absolute position for the import button? When i was creating a doc nav links i had a conversation with Irina and she said it looks better.

**Comment by QSdmitrioul** [comment]
> Screenshot 2022-05-05 at 11 34 28
> 
> Screenshot 2022-05-05 at 11 34 42
> 
> Maybe we should get back an absolute position for the import button? When i was creating a doc nav links i had a conversation with Irina and she said it looks better.

It had some bugs in realization, and it looks not very useful when button close naming of dateset, moreover it didn't give us much, I think @AndreyUstyumenko should decide.

File: src/components/input-search/input-search.tsx (hunk 10 above)
Comment (QSdmitrioul) [reply src/components/input-search/input-search.tsx]: fixed

File: src/components/input-search/input-search.module.css (hunk 1 above)
Comment (QSdmitrioul) [reply src/components/input-search/input-search.module.css]: fix

**Comment by QSdmitrioul** [comment]
> please reorganize the folder structure:
> 
> ```
> main
>  - components
>    - datasets
>        - ...components that are related to datasets
>    - selected-dataset
>       - ...components that are related to selected-dataset
>       - modals
>  - page.tsx
> ```
> 
> ui folder should be gone components that have stores must be in a separate folder

Done

**Comment by AndreyUstyumenko** [comment]
> > Screenshot 2022-05-05 at 11 34 28
> > Screenshot 2022-05-05 at 11 34 42
> > Maybe we should get back an absolute position for the import button? When i was creating a doc nav links i had a conversation with Irina and she said it looks better.
> 
> It had some bugs in realization, and it looks not very useful when button close naming of dateset, moreover it didn't give us much, I think @AndreyUstyumenko should decide.

Both options no good... I didn't like the prev version, yours is slightly better, but takes a lot of extra space under a relatively small button. Can you just place it here?
Screen Shot 2022-05-06 at 02 47 13

**Comment by QSdmitrioul** [comment]
> > > Screenshot 2022-05-05 at 11 34 28
> > > Screenshot 2022-05-05 at 11 34 42
> > > Maybe we should get back an absolute position for the import button? When i was creating a doc nav links i had a conversation with Irina and she said it looks better.
> > 
> > 
> > It had some bugs in realization, and it looks not very useful when button close naming of dateset, moreover it didn't give us much, I think @AndreyUstyumenko should decide.
> 
> Both options no good... I didn't like the prev version, yours is slightly better, but takes a lot of extra space under a relatively small button. Can you just place it here? Screen Shot 2022-05-06 at 02 47 13

Done

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**